From e78451165ac5cb4381c7cc756b138a6f55f36300 Mon Sep 17 00:00:00 2001
From: Emmanuel Gil Peyrot <linkmauve@linkmauve.fr>
Date: Mon, 3 Feb 2025 17:31:15 +0100
Subject: [PATCH] gdk/wayland: Use forgotten include guards
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

On Haiku, neither sys/sysmacros.h nor sys/syscall.h exist.  The concept
of major/minor for devices doesn’t exist, and syscall() is provided by
the system library so they aren’t useful.

These two headers were already checked by meson so no build system
change was needed.
---
 gdk/wayland/gdkdisplay-wayland.c | 5 ++++-
 gdk/wayland/gdkdmabuf-wayland.c  | 8 ++++++++
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/gdk/wayland/gdkdisplay-wayland.c b/gdk/wayland/gdkdisplay-wayland.c
index bdc74677231..7ef6f10cc67 100644
--- a/gdk/wayland/gdkdisplay-wayland.c
+++ b/gdk/wayland/gdkdisplay-wayland.c
@@ -24,14 +24,17 @@
 #include <errno.h>
 #include <unistd.h>
 #include <fcntl.h>
+
+#ifdef HAVE_SYS_SYSMACROS_H
 #include <sys/sysmacros.h>
+#endif
 
 #ifdef HAVE_LINUX_MEMFD_H
 #include <linux/memfd.h>
+#include <sys/syscall.h>
 #endif
 
 #include <sys/mman.h>
-#include <sys/syscall.h>
 
 #include <glib.h>
 #include <gio/gio.h>
diff --git a/gdk/wayland/gdkdmabuf-wayland.c b/gdk/wayland/gdkdmabuf-wayland.c
index 15d2666259c..570bad71ca1 100644
--- a/gdk/wayland/gdkdmabuf-wayland.c
+++ b/gdk/wayland/gdkdmabuf-wayland.c
@@ -9,7 +9,10 @@
 
 #include <fcntl.h>
 #include <sys/mman.h>
+
+#ifdef HAVE_SYS_SYSMACROS_H
 #include <sys/sysmacros.h>
+#endif
 
 #include "linux-dmabuf-unstable-v1-client-protocol.h"
 
@@ -52,19 +55,24 @@ update_dmabuf_formats (DmabufFormatsInfo *info)
 
   GDK_DISPLAY_DEBUG (info->display, MISC,
                      "dmabuf format table (%" G_GSIZE_FORMAT " entries)", info->n_dmabuf_formats);
+
+#ifdef HAVE_SYS_SYSMACROS_H
   GDK_DISPLAY_DEBUG (info->display, MISC,
                      "dmabuf main device: %u %u",
                      major (formats->main_device),
                      minor (formats->main_device));
+#endif
 
   for (gsize i = 0; i < formats->tranches->len; i++)
     {
       DmabufTranche *tranche = g_ptr_array_index (formats->tranches, i);
 
+#ifdef HAVE_SYS_SYSMACROS_H
       GDK_DISPLAY_DEBUG (info->display, MISC,
                          "dmabuf tranche target device: %u %u",
                          major (tranche->target_device),
                          minor (tranche->target_device));
+#endif
 
       GDK_DISPLAY_DEBUG (info->display, MISC,
                          "dmabuf%s tranche (%" G_GSIZE_FORMAT " entries):",
-- 
GitLab

